YourEasyRent/DataBase/ProductRepository.cs

```diff
@@ -7,77 +7,108 @@
 
         }
 
-        // метод GET
+
         public async Task<IEnumerable<Product>> GetProducts()
         {
-            return await _product.Find(_ => true).ToListAsync();
+            return await _productCollection.Find(_ => true).ToListAsync();
         }
         public async Task<Product> Get(string id)
         {
-            return await _product.Find(_ => _.Id == id).FirstOrDefaultAsync();
+            return await _productCollection.Find(_ => _.Id == id).FirstOrDefaultAsync();
         }
 
         public async Task<IEnumerable<Product>> GetByBrand(string brand)
         {
-            return await _product.Find(_ => _.Brand == brand).ToListAsync(); // обращеемся к объекту _ и используем его своейство Brand, где свойство нашего объекта Brand будет равно параметру Brand
+            return await _productCollection.Find(_ => _.Brand == brand).ToListAsync(); // обращеемся к объекту _ и используем его своейство Brand, где свойство нашего объекта Brand будет равно параметру Brand
         }
-       
+
 
         public async Task<Product> GetByName(string name)
         {
-            return await _product.Find(_ => _.Name == name).FirstOrDefaultAsync();
+            return await _productCollection.Find(_ => _.Name == name).FirstOrDefaultAsync();
         }
 
-        //метод POST
+
         public async Task Create(Product newproduct)
         {
-            await _product.InsertOneAsync(newproduct);  
+            await _productCollection.InsertOneAsync(newproduct);
         }
 
         public async Task<IEnumerable<string>> CreateMany(IEnumerable<Product> products)
         {
-            await _product.InsertManyAsync(products);
+            await _productCollection.InsertManyAsync(products);
 
             var createdIds = products.Select(p => p.Id); // - Здесь создается коллекция идентификаторов продуктов, используя LINQ-запрос.
 
             return createdIds;
         }
-       
-        // метод PUT
 
-        public async Task<bool> Update(string id, Product updateProduct) // метод Update возвращает значение типа bool, указывающее на успешность операции обновления. Если операция обновления прошла успешно, метод вернет true
+
+
+        public async Task<bool> Update(string brand, string name, Product updateProduct) // метод Update возвращает значение типа bool, указывающее на успешность операции обновления. Если операция обновления прошла успешно, метод вернет true
         {
-            var filter = Builders<Product>.Filter.Eq(_ => _.Id, id); // создаю фильтр для поиска продукта по его идентификатору.
-            var updateResult = await _product.ReplaceOneAsync(filter, updateProduct);// выполняет замену документа с учетом заданного фильтра.
+            var filter = Builders<Product>.Filter.And
+                (Builders<Product>.Filter.Eq(_ => _.Brand, brand),
```

Comment:
Nice job, good enough. But I'll also show you how to do it, so it works really fast.